File: modular_exponential/modular_exponential.py
def modular_exponential(base, power, modulus):

	if base > modulus:
		base %= modulus

	result = 1

	while power > 0:
		if power & 1:
			result = (result * base) % modulus
		power = power >> 1
		base = (base * base) % modulus
	return result


def main():
	print(modular_exponential(24, 2015, 1000))

if __name__ == '__main__':
	main()

File: modular_exponential/modular_exponential.java
public class modular_exponential {

	public static int modularExponential(int base, int power, int modulus) {

		if (base > modulus)
			base %= modulus;

		int result = 1;

		while (power > 0) {
			if (power % 2 == 1)
				result = (result * base) % modulus;
			power = power >> 1;
			base = (base * base) % modulus;
		}

		return result;
	}

	public static void main(String[] args) {
		System.out.println(modularExponential(24, 2015, 1000));
	}
}
